spec/matrix_spec.rb

```diff
@@ -0,0 +1,62 @@
+describe SymEngine::DenseMatrix do
+  describe 'convert' do
+    subject { SymEngine([[1, 2],[3, 4]]) }
+    
+    it { is_expected.to be_a SymEngine::DenseMatrix }
+    its(:to_s) { is_expected.to eq "[1, 2]\n[3, 4]\n" }
+  end
+  
+  describe 'dense_matrix functions' do
+    subject { SymEngine([[4, 3],[3, 2]]) }
+    
+    its(:inv) { is_expected.to be_a SymEngine::DenseMatrix }
+    its(:inv) { is_expected.to eq SymEngine([[-2, 3],[3, -4]]) }
+    
+    its(:FFLU) { is_expected.to be_a SymEngine::DenseMatrix }
+    its(:FFLU) { is_expected.to eq SymEngine([[4, 3],[3, -1]]) }
+    
+    its(:LU) { is_expected.to eq [SymEngine([[1, 0],[SymEngine(3)/SymEngine(4), 1]]), 
+                                  SymEngine([[4, 3],[0, SymEngine(-1)/SymEngine(4)]])] }
+                                  
+    its(:LDL) { is_expected.to eq [SymEngine([[1, 0],[SymEngine(3)/SymEngine(4), 1]]), 
+                                  SymEngine([[4, 0],[0, SymEngine(-1)/SymEngine(4)]])] }
+                                  
+    its(:FFLU) { is_expected.to eq SymEngine([[4, 3],[3, -1]]) }
+    
+    its(:FFLDU) { is_expected.to eq [SymEngine([[4, 0],[3, 1]]), SymEngine([[4, 0],[0, 4]]), 
+                                     SymEngine([[4, 3],[0, -1]])] }
+                                     
+    its(:det) { is_expected.to eq (-1)}  
+    
+  end
+  
+  describe 'more dense_matrix functions' do
+    
+    let(:mat1) { SymEngine([[1, 2],[3, 4]]) }
+    let(:mat2) { SymEngine([[4, 3],[2, 1]]) }
+    let(:a) { SymEngine(4) }    
+    let(:matA) { SymEngine([[3, 1, 2],[5, 7, 5], [1, 2, 3]]) }
+    let(:b) { SymEngine([[4],[4],[4]]) }
+    
+    it do
```

Comment:
Only one-line `it`s are OK to leave without descriptions, multi-line ones should be like `it 'is blah blah' do` (and when you can't invent good description, it could be a sign of wrong examples grouping).